zipline/algorithm.py

```diff
@@ -611,16 +612,28 @@ def run(self, data=None, overwrite_sim_params=True):
 
             if isinstance(data, pd.Panel):
                 # For compatibility with existing examples allow start/end
-                # to be inferred.
+                # to be inferred. Start and end have to be midnight UTC.
                 if overwrite_sim_params:
-                    self.sim_params.period_start = data.major_axis[0]
-                    self.sim_params.period_end = data.major_axis[-1]
+                    self.sim_params.period_start = normalize_date(
+                        data.major_axis[0].tz_convert('UTC')
```

Comment:
Sure. Not sure what the best way is. Could call `normalize_date` and then set the `tz` property of the result to `'UTC'`.

A tz-naive index breaks other things as well, though. May be best to automatically convert index to UTC midnight if it's daily mode.